File: 백준/Silver/2607. 비슷한 단어/비슷한 단어.py
n = int(input())
words = []
for i in range(n):
    if i == 0:
        stan = str(input())
    else:
        word = str(input())
        words.append(word)
count = 0
for i in words:
    if len(i) == len(stan):     # 문자열 길이가 같을 때
        e = sorted(stan)
        f = sorted(i)
        if e == f:     # 순서만 바꿔놓은 경우
            count += 1
        else:       # 문자 하나를 바꿔놓은 경우
            for j in range(len(e)):
                if e[j] in f:
                    f.remove(e[j])
            if len(f) == 1:
                count += 1
    elif len(i)-len(stan) == 1:    # 문자를 하나 더 추가한 경우
        a = sorted(i)
        b = sorted(stan)
        a = ''.join(str(_) for _ in a)
        b = ''.join(str(_) for _ in b)
        state = False
        for j in range(len(a)):
            if a[:j] + a[j+1:] == b:
                state = True
        if state:
            count += 1
    elif len(stan)-len(i) == 1:     # 문자를 하나 제거한 경우
        c = sorted(i)
        d = sorted(stan)
        c = ''.join(str(_) for _ in c)
        d = ''.join(str(_) for _ in d)
        state = False
        for j in range(len(d)):
            if d[:j] + d[j+1:] == c:
                state = True
        if state:
            count += 1
print(count)

File: 백준/Silver/2607. 비슷한 단어/비슷한 단어.c
#include <stdio.h>
#include <string.h>

#define MAX_LEN 101

int main() {
    int n, count = 0;
    char target[MAX_LEN], word[MAX_LEN];
    
    scanf("%d", &n);
    scanf("%s", target);
    
    for (int i = 0; i < n - 1; i++) {
        scanf("%s", word);
        
        int target_freq[26] = {0}, word_freq[26] = {0};
        
        for (int j = 0; target[j] != '\0'; j++) {
            target_freq[target[j] - 'A']++;
        }
        
        for (int j = 0; word[j] != '\0'; j++) {
            word_freq[word[j] - 'A']++;
        }
        
        int add = 0, remove = 0;
        for (int j = 0; j < 26; j++) {
            if (target_freq[j] > word_freq[j]) {
                remove += target_freq[j] - word_freq[j];
            } else {
                add += word_freq[j] - target_freq[j];
            }
        }
        
        if (add < 2 && remove < 2) {
            count++;
        }
    }
    
    printf("%d\n", count);
    return 0;
}
